pkg/probe/exec/exec.go

```diff
@@ -66,6 +68,16 @@ func (pr execProber) Probe(e exec.Cmd) (probe.Result, string, error) {
 			}
 			return probe.Failure, string(data), nil
 		}
+
+		timeoutErr, ok := err.(*TimeoutError)
+		if ok {
+			if utilfeature.DefaultFeatureGate.Enabled(features.ExecProbeTimeout) {
+				return probe.Failure, string(data), nil
+			}
+
+			klog.Warningf("Exec probe timed out after %s but ExecProbeTimeout feature gate was disabled", timeoutErr.Timeout())
```

Comment:
Why are we surfacing this as a log warning and not in a Prometheus metric?

Reply:
Couple reasons:
* this log warning is only going to be around for 1 release until we always enforce probe timeouts in v1.21., it wouldn't make sense to add a prometheus metric now and then start a deprecation period for it one release later.
* there is an action tied to this warning and hints that the user should not be relying on this behavior going forward -- this would be hard to convey in a prometheus metric.

Reply:
> it wouldn't make sense to add a prometheus metric now and then start a deprecation period for it one release later.

FWIW this is the point of alpha metrics :) see: https://github.com/kubernetes/enhancements/blob/master/keps/sig-instrumentation/20190404-kubernetes-control-plane-metrics-stability.md#stability-classes